content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md

```diff
@@ -0,0 +1,488 @@
+
+Suppose that you configured a
+[`serviceMonitorSelector`](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#opentelemetrycollectorspectargetallocatorprometheuscr-1)
+for your Target Allocator, like in the following example:
+
+```yaml
+apiVersion: opentelemetry.io/v1beta1
+kind: OpenTelemetryCollector
+metadata:
+  name: otelcol
+  namespace: opentelemetry
+spec:
+  mode: statefulset
+  targetAllocator:
+    enabled: true
+    serviceAccount: opentelemetry-targetallocator-sa
+    prometheusCR:
+      enabled: true
+      serviceMonitorSelector:
+        matchLabels:
+          app: my-app
+```
+
+By setting the value of
+`spec.targetAllocator.prometheusCR.serviceMonitorSelector.matchLabels` to
+`app: my-app`, it means that your `ServiceMonitor` resource must in turn have
+that same value in `metadata.labels`:
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: sm-example
+  labels:
+    app: my-app
+    release: prometheus
+spec:
+```
+
+See the full `ServiceMonitor`
+[resource definition in "Do you know if metrics are actually being scraped?"](#do-you-know-if-metrics-are-actually-beingscraped).
+
+In this case, the `OpenTelemetryCollector` resource's
+`prometheusCR.serviceMonitorSelector.matchLabels` is looking only for
+`ServiceMonitors` having the label `app: my-app`, which we see in the previous
+example.
+
+If your `ServiceMonitor` resource is missing that label, then the Target
+Allocator will fail to discover scrape targets from that `ServiceMonitor`.
+
+{{% alert title="Tip" %}}
+
+The same applies if you’re using a
+[PodMonitor](https://prometheus-operator.dev/docs/user-guides/getting-started/#using-podmonitors).
+In that case, you would use a
+[`podMonitorSelector`](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#opentelemetrycollectorspectargetallocatorprometheuscr)
+instead of a `serviceMonitorSelector`.
+
+{{% /alert %}}
+
+### Did you leave out the serviceMonitorSelector and/or podMonitorSelector configuration altogether?
+
+As mentioned in
+["Did you configure a ServiceMonitor or PodMonitor selector"](#did-you-configure-a-servicemonitor-or-podmonitor-selector),
+setting mismatched values for `serviceMonitorSelector` and `podMonitorSelector`
+results in the Target Allocator failing to discover scrape targets from your
+`ServiceMonitors` and `PodMonitors`, respectively.
+
+Similarly, in
+[`v1beta1`](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#opentelemetrycollector-1)
+of the `OpenTelemetryCollector` CR, leaving out this configuration altogether
+also results in the Target Allocator failing to discover scrape targets from
+your `ServiceMonitors` and `PodMonitors`.
+
+As of `v1beta1` of the `OpenTelemetryOperator`, a `serviceMonitorSelector` and
+`podMonitorSelector` must be included, even if you don’t intend to use it, like
+this:
```

Comment:
i would include that setting it like this means it will match everything.

Reply:
Done!